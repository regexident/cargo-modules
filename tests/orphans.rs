#[macro_use]
mod util;

mod help {
    test_cmd!(
        args: "orphans \
                --help",
        success: true,
        color_mode: ColorMode::Plain,
        project: smoke
    );
}

mod colors {
    mod plain {
        test_cmd!(
            args: "orphans",
            success: false,
            color_mode: ColorMode::Plain,
            project: orphans
        );
    }

    mod ansi {
        test_cmd!(
            args: "orphans",
            success: false,
            color_mode: ColorMode::Ansi,
            project: orphans
        );
    }

    mod truecolor {
        test_cmd!(
            args: "orphans",
            success: false,
            color_mode: ColorMode::TrueColor,
            project: orphans
        );
    }
}

mod no_orphans {
    test_cmd!(
        args: "orphans",
        success: true,
        color_mode: ColorMode::Plain,
        project: no_orphans
    );
}

mod orphans {
    test_cmd!(
        args: "orphans",
        success: false,
        color_mode: ColorMode::Plain,
        project: orphans
    );
}
