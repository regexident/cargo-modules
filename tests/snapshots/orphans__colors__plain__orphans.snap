---
source: tests/orphans.rs
expression: output
---
STDERR:
Error: Found 2 orphans in crate 'orphans'

STDOUT:

2 orphans found:

warning: orphaned module `bar` at src/orphans/bar.rs
  --> src/orphans.rs
   |  ^^^^^^^^^^^^^^ orphan module not loaded from file
   |
 help: consider loading `bar` from module `orphans::orphans`
   |
   |  mod bar;
   |  ++++++++
   |

warning: orphaned module `foo` at src/orphans/foo/mod.rs
  --> src/orphans.rs
   |  ^^^^^^^^^^^^^^ orphan module not loaded from file
   |
 help: consider loading `foo` from module `orphans::orphans`
   |
   |  mod foo;
   |  ++++++++
   |
